datalab/datalab_session/util.py

```diff
@@ -97,6 +98,30 @@ def get_archive_from_basename(basename: str) -> dict:
 
   return results
 
+def get_hdu(basename: str, extension: str = 'SCI') -> list[fits.HDUList]:
+  """
+  Returns a list of Sci HDUs for the given basenames
+  Warning: this function returns an opened file that must be closed after use
+  """
+
+  # use the basename to fetch and create a list of hdu objects
+  basename = basename.replace('-large', '').replace('-small', '')
+
+  if cache.get(f'{basename}-{extension}') is not None:
+    return cache.get(f'{basename}-{extension}')
+
+  archive_record = get_archive_from_basename(basename)
+
+  try:
+    fits_url = archive_record[0].get('url', 'No URL found')
+  except IndexError:
+    RuntimeWarning(f"No image found with specified basename: {basename}")
+
+  # not sure if HDU is ever being closed here?
+  hdu = fits.open(fits_url, use_fsspec=True)
```

Comment:
I don't think it is - using a context manager to open the frame and get the correct data in memory doesn't work?

Reply:
I left a comment under your suggestion for it, let me add a tag if I can